### DISCUSSION TITLE
怎么升级到Taro4.0呀?

### DISCUSSION THREAD

**Discussion by ice909** [NervJS/taro#16217] [Q & A]
直接改版本号删除pnpm-lock,node_modules/* 升级不行

**Comment by viilon** [comment]
taro update project

**Reply by ice909** [reply]
这么升级上去项目跑不起来
```bash
✖ Errors: 

ModuleBuildError: Module build failed (from ./node_modules/.pnpm/postcss-loader@8.1.1_postcss@6.0.23_typescript@4.1.2_webpack@5.78.0_@swc+core@1.3.96_/node_modules/postcss-loader/dist/cjs.js):
Error: [object Object] is not a PostCSS plugin

→ Watching... [2024/8/2 09:03:29]

node:internal/process/promises:289
            triggerUncaughtException(err, true /* fromPromise */);
            ^

[UnhandledPromiseRejection: This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason "[object Array]".] {
  code: 'ERR_UNHANDLED_REJECTION'
}

Node.js v20.14.0
 ELIFECYCLE  Command failed with exit code 1.
```

**Comment by honlyHuang** [comment]
[文档](https://docs.taro.zone/docs/GETTING-STARTED#%E4%BF%9D%E6%8C%81-cli-%E7%9A%84%E7%89%88%E6%9C%AC%E4%B8%8E%E5%90%84%E7%AB%AF%E4%BE%9D%E8%B5%96%E7%89%88%E6%9C%AC%E4%B8%80%E8%87%B4)
```
# 使用Taro 升级命令更新CLI版本到最新版本
$ taro update self
# 使用Taro 升级命令更新CLI版本到指定版本
$ taro update self [版本号]
```

**Reply by ice909** [reply]
> [文档](https://docs.taro.zone/docs/GETTING-STARTED#%E4%BF%9D%E6%8C%81-cli-%E7%9A%84%E7%89%88%E6%9C%AC%E4%B8%8E%E5%90%84%E7%AB%AF%E4%BE%9D%E8%B5%96%E7%89%88%E6%9C%AC%E4%B8%80%E8%87%B4)
> 
> ```
> # 使用Taro 升级命令更新CLI版本到最新版本
> $ taro update self
> # 使用Taro 升级命令更新CLI版本到指定版本
> $ taro update self [版本号]
> ```

![image](https://github.com/user-attachments/assets/eee6e37f-6b58-4069-b57d-7169f389e2c8)
升级报错了
